Add Mapit Areas JSON scraper

chrismytton opened the pull request:
This scrapes http://global.mapit.mysociety.org/areas/OCL to get a list of French constituencies from Mapit.

Part of https://github.com/everypolitician/everypolitician/issues/595

chrismytton committed:
Add Mapit Areas JSON scraper

tmtmtmtm reviewed (commented):
This feels like one where we want to create a `Scraped::JSON` subclass, even if just a partial implementation that's Good Enough For Now™. If `Scraped` already had such a thing, then we'd definitely be using it here, but it's never going to actually get one until something (like this) needs it and so pushes us towards it.

I don't think we necessarily need to implement it within `Scraped` itself yet, but having a bare-bones version here that could be the basis for pushing it up to core later is probably worth it. (I'm open to persuasion that that's going to be too much effort here, when we really want to get this released, so should be a refactoring we do Real Soon Now™, but my suspicion is that it'd actually be fairly quick)

What do you think?

chrismytton committed:
Add Scraped::JSON class

This can be used as a prototype before we actually add this class into
Scraped. This class provides a `json` method which returns the response
body JSON, or if the `fragment` method was used it returns the subset of
the response JSON passed to `fragment`.

chrismytton commented:
> This feels like one where we want to create a Scraped::JSON subclass, even if just a partial implementation that's Good Enough For Now™.

Yep, good idea. I did consider doing that originally but thought it would take too long. However I've just had a crack at it and it was pleasantly simple. Have pushed https://github.com/everypolitician-scrapers/france-constituencies-mapit/pull/1/commits/27c650b531ec9ad0fddca466e80d1e4aa775069b with that change.

tmtmtmtm reviewed (approved):
I hesitated a bit over this because of the lack of tests, especially as we know this is going to need to change again real soon now (when the `ref` field is added), and because the principle of `ensure_regression_tests` works best when tests are there from the very start. So this is pushing much more work onto whomever makes that next change.

The implementation of `Scraped::JSON` also gives me some pause, and I suspect integrating this into the core of `Scraped` isn't going to be as simple as we would like.

But due to the time pressures on this, I guess it's good enough for now.
